### PR TITLE
Pr multi gpu

### PR DESCRIPTION
**Pull request by Haichao-Zhang**

### CODE REVIEW

File: alf/utils/multi_gpu_utils_test.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            outside_batch_size = input.size(0)
+
+            self.assertTrue(in_model_batch_size * len(gpu_ids),
+                            outside_batch_size)
+
+
+class TestMultiGPUCustomizedFunction(unittest.TestCase):
+    def test_multi_gpu_customized_function(self):
+        device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        model = Model(input_size, output_size).to(device)
```
Comment (hnyu) [inline alf/utils/multi_gpu_utils_test.py]: Do we always have to first put model on one of the gpus?

Comment (Haichao-Zhang) [reply alf/utils/multi_gpu_utils_test.py]: yes

File: alf/utils/multi_gpu_utils_test.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    def test_run_on_cpu(self):
+        model = Model(input_size, output_size)  # model is on cpu
+
+        model = MultiGPU(model)
+        data_loader = DataLoader(
+            dataset=DummyDataset(input_size, data_size),
+            batch_size=batch_size,
+            shuffle=True)
+
+        # check model is till on cpu
```
Comment (hnyu) [inline alf/utils/multi_gpu_utils_test.py]: till -> still

**Comment by hnyu** [comment]
Do you have any plan to integrate this feature into the trainer framework?

**Comment by Haichao-Zhang** [comment]
Close for now and reopen in a future point when necessary.

**Pull request closed without merging**